src/structure.js

```diff
@@ -280,8 +276,7 @@ Structure.prototype.reference = function (path) {
      * @returns {Void}
      */
     destroy: function () {
-      removeAllListenersBut(self, pathId, referenceListeners);
-      referenceListeners = void 0;
+      removeAllListenersBut(listenerNs);
```

Comment:
Similar to https://github.com/omniscientjs/immstruct/pull/56/files#r28195201. This destroys all observers attached by other cursor references of the same keypath; not only its own.